src/Dialogs/Waypoint/WaypointList.cpp

```
@@ -38,14 +38,38 @@
 #include <list>
 
 #include <cassert>
 #include <stdio.h>
 
+/**
+ * Calculate the reachability of the given waypoint the same way the
+ * map does, so the icon matches the one on the map.
+ */
+static WaypointReachability
+GetWaypointReachability(const Waypoint &waypoint) noexcept
+{
```

Cppcheck (2.21.0)

[error] Reference to temporary returned.

(returnTempReference)